kore/test/Test/Kore/Unification/SubstitutionNormalization.hs

```diff
@@ -24,100 +28,172 @@ import qualified Test.Kore.Step.MockSymbols as Mock
+                , (RegVar $ x1 Mock.testSort, mkTop Mock.testSort)
                 ]
     , testCase "Unnormalized substitution with 'and'" $
         assertEqualWithExplanation ""
-            (Right
-                [   ( v1 Mock.testSort
+            (Substitution
+                [   ( RegVar $ v1 Mock.testSort
                     , mkAnd mkTop_ (mkTop Mock.testSort)
                     )
-                , (x1 Mock.testSort, mkTop Mock.testSort)
+                , (RegVar $ x1 Mock.testSort, mkTop Mock.testSort)
                 ]
             )
             =<< runNormalizeSubstitution
-                [   ( v1 Mock.testSort
+                [   (RegVar $ v1 Mock.testSort
                     , mkAnd (mkVar $ x1 Mock.testSort) mkTop_
                     )
-                ,   (x1 Mock.testSort, mkTop Mock.testSort)
+                ,   (RegVar $ x1 Mock.testSort, mkTop Mock.testSort)
                 ]
     , testCase "Simplest cycle" $ do
         let var1 = v1 Mock.testSort
-        assertEqualWithExplanation "" (Right [])
-            =<< runNormalizeSubstitution [(var1, mkVar $ v1 Mock.testSort)]
+        assertEqualWithExplanation "" (Substitution [])
+            =<< runNormalizeSubstitution [(RegVar var1, mkVar $ v1 Mock.testSort)]
     , testCase "Cycle with extra substitution" $ do
         let
             var1 = v1 Mock.testSort
             varx1 = x1 Mock.testSort
-        assertEqualWithExplanation "" (Right [(varx1, mkVar var1)])
+        assertEqualWithExplanation "" (Substitution [(RegVar varx1, mkVar var1)])
             =<< runNormalizeSubstitution
-                    [ (var1, mkVar var1)
-                    , (varx1, mkVar var1)
+                    [ (RegVar var1, mkVar var1)
+                    , (RegVar varx1, mkVar var1)
+                    ]
+    , testCase "SetVariable Simplest cycle" $ do
+        let var1 = Mock.makeTestSubstVar "@x"
+        assertEqualWithExplanation "" (Substitution [])
+            =<< runNormalizeSubstitution [(var1, mkSubstVar var1)]
+    , testCase "SetVariable Cycle with extra substitution" $ do
+        let
+            var1 = Mock.makeTestSubstVar "@v"
+            varx1 = Mock.makeTestSubstVar "@x"
+        assertEqualWithExplanation "" (Substitution [(varx1, mkSubstVar var1)])
+            =<< runNormalizeSubstitution
+                    [ (var1, mkSubstVar var1)
+                    , (varx1, mkSubstVar var1)
                     ]
     , testCase "Function cycle" $ do
         let var1 = v1 Mock.testSort
         assertEqualWithExplanation ""
-            (Left (NonCtorCircularVariableDependency [var1]))
+            (Error (NonCtorCircularVariableDependency [RegVar var1]))
             =<< runNormalizeSubstitution
-                [ ( var1 , mkApplySymbol f [mkVar var1] ) ]
-    , testCase "Length 2 cycle" $ do
+                [ (RegVar  var1 , mkApplySymbol f [mkVar var1] ) ]
+    {- TODO(traiansf): maybe check that exception is thrown here
```

Comment:
You could actually do the test, it's just an `assertError` (see Test.Tasty.HUnit.Extensions). Getting an error is not that important in itself, but you would be checking that we're not doing something stupid instead.